src/material/core/tokens/m2/mdc/_button-filled.scss

```diff
@@ -0,0 +1,92 @@
+@use '../../token-utils';
+@use '../../../mdc-helpers/mdc-helpers';
+@use '../../../style/sass-utils';
+@use '../../../theming/inspection';
+
+// The prefix used to generate the fully qualified name for tokens in this file.
+$prefix: (mdc, button-filled);
+
+// Tokens that can't be configured through Angular Material's current theming API,
+// but may be in a future version of the theming API.
+//
+// Tokens that are available in MDC, but not used in Angular Material should be mapped to `null`.
+// `null` indicates that we are intentionally choosing not to emit a slot or value for the token in
+// our CSS.
+@function get-unthemable-tokens() {
+  @return (
+    container-elevation: 0,
+    container-height: 36px,
+    container-shape: 4px,
+    disabled-container-elevation: 0,
+    focus-container-elevation: 0,
+    hover-container-elevation: 0,
+    keep-touch-target: false,
+    pressed-container-elevation: 0,
+
+    focus-state-layer-opacity: null,
+    hover-state-layer-opacity: null,
+    pressed-state-layer-opacity: null,
+    container-shadow-color: null,
+    focus-label-text-color: null,
+    hover-label-text-color: null,
+    label-text-font: null,
+    label-text-size: null,
+    label-text-tracking: null,
+    label-text-transform: null,
+    label-text-weight: null,
+    pressed-label-text-color: null,
+    with-icon-disabled-icon-color: null,
+    with-icon-focus-icon-color: null,
+    with-icon-hover-icon-color: null,
+    with-icon-icon-color: null,
+    with-icon-icon-size: null,
+    with-icon-pressed-icon-color: null
+  );
+}
+
+@function on-color($theme, $palette) {
+  @if ($palette) {
+    $is-dark: inspection.get-theme-type($theme) == dark;
+    @return if(mdc-helpers.variable-safe-contrast-tone($palette, $is-dark) == 'dark', #000, #fff);
+  }
+}
+
+// Tokens that can be configured through Angular Material's color theming API.
+@function get-color-tokens($theme, $color: null, $on-color: null) {
+  $is-dark: inspection.get-theme-type($theme) == dark;
+  $primary: inspection.get-theme-color($theme, primary);
+  $surface: inspection.get-theme-color($theme, background, card);
+  $on-primary: on-color($theme, $primary);
+  $on-surface: on-color($theme, $surface);
+
+  @return (
+    container-color: if($color, $color, transparent),
+    disabled-container-color: rgba($on-surface, 0.12),
+    disabled-label-text-color: rgba($on-surface, if($is-dark, 0.5, 0.38)),
+    focus-state-layer-color: $on-primary,
+    hover-state-layer-color: $on-primary,
+    label-text-color: if($on-color, $on-color, inherit),
+    pressed-state-layer-color: $on-primary
+  );
+}
+
+// Tokens that can be configured through Angular Material's typography theming API.
+@function get-typography-tokens($theme) {
+  @return ();
```

Comment:
The typography and density styles are still using MDC's old API. Shouldn't we switch them over as well?

Reply:
Yeah this is what I was trying to bring up in standup but didn't have time to dive into. Currently all button variants are relying on the same styles for density & typography. I asked if I should split those out and Andrew said he was ok with landing this without doing so

Reply:
That said, I don't really know if I was able to explain the situation in the short time we had during standup. I'll ping Andrew and see if he can weigh in here

Reply:
That's at least what we've been doing when converting the other components to tokens. For the filled button specifically it looks like we have these 5 tokens for the typography: `label-text-font`, `label-text-size`, `label-text-tracking`, `label-text-transform`, `label-text-weight` and the `container-height` token for the density.

Reply:
Had a chat with Wagner - it'll be a bit easier to land the typography changes across all the variants at once after these PRs land. It'll just be a different iterative approach that reaches the same goal in the end (remove the old MDC mixins)

Reply:
Alright, I wanted to make sure that we do it before we consider the button as migrated.